src/library/dao/playlistdao.cpp

```diff
@@ -208,6 +208,27 @@ void PlaylistDAO::deletePlaylist(const int playlistId) {
     emit deleted(playlistId);
 }
 
+void PlaylistDAO::deleteEmptyPlaylists(const PlaylistDAO::HiddenType type) {
+    //qDebug() << "PlaylistDAO::deletePlaylist" << QThread::currentThread() << m_database.connectionName();
+    ScopedTransaction transaction(m_database);
```

Comment:
Transactions help the database to correctly flush and helps with constancy. Without measuring it, I would even suspect it's faster with then without. As long as we don't have proper transaction infrastructure, I would rather let it in.

Reply:
SQLite does not support nested transactions. The transaction around a single statement is not only redundant here, it may cause an outer transaction to not work as intended.